.github/workflows/release.yaml

```
@@ -16,28 +16,29 @@ jobs:
     uses: ./.github/workflows/check.yaml
     secrets: inherit
 
+
   release:
     needs: check
-    runs-on: ${{ matrix.runs-on }}
-    strategy:
-      fail-fast: false
-      matrix:
-        runs-on: [[ubuntu-22.04]]
+    runs-on: ubuntu-24.04
     steps:
       - name: Checkout
         uses: actions/checkout@v4
         with:
           submodules: true
 
-      - name: Initialize lxd  # This should dropped once it's implemented on charming-actions itself. https://github.com/canonical/charming-actions/issues/140
-        # revision is latest main at time of writing; using because it contains a fix to https://github.com/canonical/setup-lxd/issues/19
-        uses: canonical/setup-lxd@2aa6f7caf7d1484298a64192f7f63a6684e648a4
+      - name: Download the tested charms
+        uses: actions/download-artifact@v4
+        with:
+          pattern: built_charms_*
+          merge-multiple: true
+
+      - name: List the names of the tested charms
+        run: |
+          echo "CHARM_NAMES=$(ls *.charm | paste -sd ,)" | tee -a "$GITHUB_ENV"
 
       - name: Pack and upload to charmhub
         uses: canonical/charming-actions/upload-charm@2.6.2
```

I believe comma-separated charm names for built-charm-path support was only added in v2.7.0:

- (not in 2.6.3) https://github.com/canonical/charming-actions/blob/2.6.3/upload-charm/README.md?plain=1#L37
- (is in 2.7.0) https://github.com/canonical/charming-actions/blob/2.7.0/upload-charm/README.md?plain=1#L37
- the related PR: https://github.com/canonical/charming-actions/pull/156

```suggestion
        uses: canonical/charming-actions/upload-charm@2.7.0
```

---

Thanks! I missed this one